File: feature/home/src/commonMain/kotlin/com/syrmos/feature/home/WeatherCard.kt
package com.syrmos.feature.home

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.syrmos.core.common.AppLanguage
import com.syrmos.core.model.weather.CurrentWeather
import com.syrmos.core.model.weather.WeatherCondition
import com.syrmos.core.model.weather.WeatherSnapshot
import kotlin.math.roundToInt

/**
 * Weather travel-context card, in the vibrant gradient style of the shared
 * weather-app design: a condition-tinted gradient, a big temperature, the
 * emoji glyph, "feels like", and the place. Pure surfacing of [WeatherSnapshot].
 */
@Composable
fun WeatherCard(snapshot: WeatherSnapshot, lang: AppLanguage) {
    val w = snapshot.current
    val colors = gradientFor(w.condition, w.isDay)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(Brush.verticalGradient(colors))
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Top,
        ) {
            Column {
                Text(
                    text = snapshot.placeName,
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White.copy(alpha = 0.95f),
                )
                Text(
                    text = conditionLabel(w.condition, lang),
                    style = MaterialTheme.typography.bodyMedium,
                    color = Color.White.copy(alpha = 0.85f),
                )
            }
            Text(text = glyph(w.condition, w.isDay), fontSize = 40.sp)
        }
        Spacer(Modifier.height(2.dp))
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.Bottom,
        ) {
            Text(
                text = "${w.temperatureC.roundToInt()}°",
                fontSize = 56.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
            )
            Spacer(Modifier.weight(1f))
            if (snapshot.highC != null && snapshot.lowC != null) {
                Text(
                    text = "H:${snapshot.highC!!.roundToInt()}°  L:${snapshot.lowC!!.roundToInt()}°",
                    style = MaterialTheme.typography.labelLarge,
                    color = Color.White.copy(alpha = 0.9f),
                    modifier = Modifier.padding(bottom = 8.dp),
                )
            }
        }

        if (snapshot.hourly.isNotEmpty()) {
            Spacer(Modifier.height(8.dp))
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(14.dp))
                    .background(Color.White.copy(alpha = 0.15f))
                    .padding(vertical = 10.dp),
                horizontalArrangement = Arrangement.SpaceEvenly,
            ) {
                snapshot.hourly.take(6).forEach { h ->
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text(h.hourLabel, style = MaterialTheme.typography.labelSmall, color = Color.White.copy(alpha = 0.85f))
                        Text(glyph(h.condition, w.isDay), fontSize = 18.sp)
                        Text("${h.temperatureC.roundToInt()}°", style = MaterialTheme.typography.labelMedium, fontWeight = FontWeight.SemiBold, color = Color.White)
                    }
                }
            }
        }

        Spacer(Modifier.height(8.dp))
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            WeatherStat(feelsLikeLabel(lang), "${w.apparentC.roundToInt()}°")
            WeatherStat(humidityLabel(lang), "${w.humidity}%")
            WeatherStat(windLabel(lang), "${w.windKph.roundToInt()} km/h")
        }
    }
}

@Composable
private fun WeatherStat(label: String, value: String) {
    Column {
        Text(label, style = MaterialTheme.typography.labelSmall, color = Color.White.copy(alpha = 0.8f))
        Text(value, style = MaterialTheme.typography.bodyMedium, fontWeight = FontWeight.SemiBold, color = Color.White)
    }
}

private fun gradientFor(condition: WeatherCondition, isDay: Boolean): List<Color> {
    if (!isDay) return listOf(Color(0xFF1A237E), Color(0xFF311B92))
    return when (condition) {
        WeatherCondition.CLEAR -> listOf(Color(0xFF2196F3), Color(0xFF4FC3F7))
        WeatherCondition.PARTLY_CLOUDY -> listOf(Color(0xFF42A5F5), Color(0xFF90CAF9))
        WeatherCondition.CLOUDY, WeatherCondition.FOG -> listOf(Color(0xFF607D8B), Color(0xFF90A4AE))
        WeatherCondition.DRIZZLE, WeatherCondition.RAIN, WeatherCondition.SHOWERS ->
            listOf(Color(0xFF455A64), Color(0xFF546E7A))
        WeatherCondition.THUNDERSTORM -> listOf(Color(0xFF37474F), Color(0xFF455A64))
        WeatherCondition.SNOW -> listOf(Color(0xFF78909C), Color(0xFFB0BEC5))
        WeatherCondition.UNKNOWN -> listOf(Color(0xFF42A5F5), Color(0xFF90CAF9))
    }
}

private fun glyph(condition: WeatherCondition, isDay: Boolean): String = when (condition) {
    WeatherCondition.CLEAR -> if (isDay) "☀️" else "🌙"
    WeatherCondition.PARTLY_CLOUDY -> if (isDay) "⛅" else "☁️"
    WeatherCondition.CLOUDY -> "☁️"
    WeatherCondition.FOG -> "🌫️"
    WeatherCondition.DRIZZLE, WeatherCondition.SHOWERS -> "🌦️"
    WeatherCondition.RAIN -> "🌧️"
    WeatherCondition.SNOW -> "❄️"
    WeatherCondition.THUNDERSTORM -> "⛈️"
    WeatherCondition.UNKNOWN -> "🌡️"
}

internal fun conditionLabel(condition: WeatherCondition, lang: AppLanguage): String = when (condition) {
    WeatherCondition.CLEAR -> tri(lang, "Clear", "Αίθριος", "Kthjellët", "Sereno")
    WeatherCondition.PARTLY_CLOUDY -> tri(lang, "Partly cloudy", "Λίγα σύννεφα", "Pjesërisht me re", "Parzialmente nuvoloso")
    WeatherCondition.CLOUDY -> tri(lang, "Cloudy", "Συννεφιά", "Me re", "Nuvoloso")
    WeatherCondition.FOG -> tri(lang, "Fog", "Ομίχλη", "Mjegull", "Nebbia")
    WeatherCondition.DRIZZLE -> tri(lang, "Drizzle", "Ψιχάλα", "Shi i imët", "Pioviggine")
    WeatherCondition.RAIN -> tri(lang, "Rain", "Βροχή", "Shi", "Pioggia")
    WeatherCondition.SHOWERS -> tri(lang, "Showers", "Μπόρες", "Rrebeshe", "Rovesci")
    WeatherCondition.SNOW -> tri(lang, "Snow", "Χιόνι", "Borë", "Neve")
    WeatherCondition.THUNDERSTORM -> tri(lang, "Thunderstorm", "Καταιγίδα", "Stuhi", "Temporale")
    WeatherCondition.UNKNOWN -> tri(lang, "Weather", "Καιρός", "Moti", "Meteo")
}

private fun feelsLikeLabel(lang: AppLanguage) = tri(lang, "Feels like", "Αίσθηση", "Ndihet si", "Percepita")
private fun humidityLabel(lang: AppLanguage) = tri(lang, "Humidity", "Υγρασία", "Lagështia", "Umidità")
private fun windLabel(lang: AppLanguage) = tri(lang, "Wind", "Άνεμος", "Era", "Vento")

private fun tri(lang: AppLanguage, en: String, el: String, sq: String, it: String) = when (lang) {
    AppLanguage.GREEK -> el
    AppLanguage.ALBANIAN -> sq
    AppLanguage.ITALIAN -> it
    else -> en
}

File: iosApp/iosApp/Features/Weather/WeatherCard.swift
import SwiftUI

/// Weather travel-context card, in the vibrant gradient style of the shared
/// weather-app design. Mirrors the Compose `WeatherCard`.
struct WeatherCard: View {
    let snapshot: WeatherSnapshot
    @ObservedObject private var loc = LocalizationManager.shared

    var body: some View {
        let w = snapshot.current
        VStack(alignment: .leading, spacing: 6) {
            HStack(alignment: .top) {
                VStack(alignment: .leading, spacing: 2) {
                    Text(snapshot.placeName)
                        .font(.subheadline).fontWeight(.semibold)
                        .foregroundStyle(.white.opacity(0.95))
                    Text(WeatherStyle.label(w.condition, loc.language))
                        .font(.body)
                        .foregroundStyle(.white.opacity(0.85))
                }
                Spacer()
                Text(WeatherStyle.glyph(w.condition, w.isDay)).font(.system(size: 40))
            }
            HStack(alignment: .bottom) {
                Text("\(Int(w.temperatureC.rounded()))°")
                    .font(.system(size: 56, weight: .bold))
                    .foregroundStyle(.white)
                Spacer()
                if let high = snapshot.highC, let low = snapshot.lowC {
                    Text("H:\(Int(high.rounded()))°  L:\(Int(low.rounded()))°")
                        .font(.subheadline)
                        .foregroundStyle(.white.opacity(0.9))
                        .padding(.bottom, 8)
                }
            }

            if !snapshot.hourly.isEmpty {
                HStack {
                    ForEach(snapshot.hourly.prefix(6)) { h in
                        VStack(spacing: 3) {
                            Text(h.hourLabel).font(.caption2).foregroundStyle(.white.opacity(0.85))
                            Text(WeatherStyle.glyph(h.condition, w.isDay)).font(.system(size: 18))
                            Text("\(Int(h.temperatureC.rounded()))°").font(.caption).fontWeight(.semibold).foregroundStyle(.white)
                        }
                        .frame(maxWidth: .infinity)
                    }
                }
                .padding(.vertical, 10)
                .background(.white.opacity(0.15))
                .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            }

            HStack(spacing: 16) {
                stat(feelsLike, "\(Int(w.apparentC.rounded()))°")
                stat(humidity, "\(w.humidity)%")
                stat(wind, "\(Int(w.windKph.rounded())) km/h")
            }
        }
        .padding(20)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            LinearGradient(colors: WeatherStyle.gradient(w.condition, w.isDay),
                           startPoint: .top, endPoint: .bottom)
        )
        .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
    }

    private func stat(_ label: String, _ value: String) -> some View {
        VStack(alignment: .leading, spacing: 1) {
            Text(label).font(.caption2).foregroundStyle(.white.opacity(0.8))
            Text(value).font(.body).fontWeight(.semibold).foregroundStyle(.white)
        }
    }

    private var feelsLike: String { tri("Feels like", "Αίσθηση", "Ndihet si", "Percepita") }
    private var humidity: String { tri("Humidity", "Υγρασία", "Lagështia", "Umidità") }
    private var wind: String { tri("Wind", "Άνεμος", "Era", "Vento") }

    private func tri(_ en: String, _ el: String, _ sq: String, _ it: String? = nil) -> String {
        switch loc.language { case .greek: return el; case .albanian: return sq; case .italian: return it ?? en; default: return en }
    }
}

/// Shared weather visuals so the card and any future weather surface agree.
enum WeatherStyle {
    static func glyph(_ c: WeatherCondition, _ isDay: Bool) -> String {
        switch c {
        case .clear: return isDay ? "☀️" : "🌙"
        case .partlyCloudy: return isDay ? "⛅" : "☁️"
        case .cloudy: return "☁️"
        case .fog: return "🌫️"
        case .drizzle, .showers: return "🌦️"
        case .rain: return "🌧️"
        case .snow: return "❄️"
        case .thunderstorm: return "⛈️"
        case .unknown: return "🌡️"
        }
    }

    static func gradient(_ c: WeatherCondition, _ isDay: Bool) -> [Color] {
        if !isDay { return [Color(red: 0.10, green: 0.14, blue: 0.49), Color(red: 0.19, green: 0.11, blue: 0.57)] }
        switch c {
        case .clear: return [Color(red: 0.13, green: 0.59, blue: 0.95), Color(red: 0.31, green: 0.76, blue: 0.97)]
        case .partlyCloudy: return [Color(red: 0.26, green: 0.65, blue: 0.96), Color(red: 0.56, green: 0.79, blue: 0.98)]
        case .cloudy, .fog: return [Color(red: 0.38, green: 0.49, blue: 0.55), Color(red: 0.56, green: 0.64, blue: 0.68)]
        case .drizzle, .rain, .showers: return [Color(red: 0.27, green: 0.35, blue: 0.39), Color(red: 0.33, green: 0.43, blue: 0.48)]
        case .thunderstorm: return [Color(red: 0.22, green: 0.28, blue: 0.31), Color(red: 0.27, green: 0.35, blue: 0.39)]
        case .snow: return [Color(red: 0.47, green: 0.56, blue: 0.61), Color(red: 0.69, green: 0.75, blue: 0.77)]
        case .unknown: return [Color(red: 0.26, green: 0.65, blue: 0.96), Color(red: 0.56, green: 0.79, blue: 0.98)]
        }
    }

    static func label(_ c: WeatherCondition, _ lang: AppLanguage) -> String {
        func quad(_ en: String, _ el: String, _ sq: String, _ it: String) -> String {
            switch lang { case .greek: return el; case .albanian: return sq; case .italian: return it; default: return en }
        }
        switch c {
        case .clear: return quad("Clear", "Αίθριος", "Kthjellët", "Sereno")
        case .partlyCloudy: return quad("Partly cloudy", "Λίγα σύννεφα", "Pjesërisht me re", "Parzialmente nuvoloso")
        case .cloudy: return quad("Cloudy", "Συννεφιά", "Me re", "Nuvoloso")
        case .fog: return quad("Fog", "Ομίχλη", "Mjegull", "Nebbia")
        case .drizzle: return quad("Drizzle", "Ψιχάλα", "Shi i imët", "Pioggerella")
        case .rain: return quad("Rain", "Βροχή", "Shi", "Pioggia")
        case .showers: return quad("Showers", "Μπόρες", "Rrebeshe", "Rovesci")
        case .snow: return quad("Snow", "Χιόνι", "Borë", "Neve")
        case .thunderstorm: return quad("Thunderstorm", "Καταιγίδα", "Stuhi", "Temporale")
        case .unknown: return quad("Weather", "Καιρός", "Moti", "Meteo")
        }
    }
}
